cli/src/commands/git/push.rs

```diff
@@ -295,8 +300,39 @@ pub fn cmd_git_push(
         return Ok(());
     }
 
-    validate_commits_ready_to_push(ui, &bookmark_updates, &remote, &tx, args)?;
+    let commits_to_sign =
+        validate_commits_ready_to_push(ui, &bookmark_updates, &remote, &tx, args)?;
+    let num_updated_signatures = commits_to_sign.len();
+    let (num_rebased_descendants, bookmark_updates) =
+        sign_commits_before_push(&mut tx, bookmark_updates, commits_to_sign, args.dry_run)?;
     if let Some(mut formatter) = ui.status_formatter() {
+        if num_updated_signatures > 0 {
+            if args.dry_run {
+                writeln!(
+                    formatter,
+                    "Dry-run requested, skipped updating signatures of {num_updated_signatures} \
+                     commits"
+                )?;
+                if num_rebased_descendants > 0 {
+                    writeln!(
+                        formatter,
+                        "Dry-run requested, skipped rebasing of {num_rebased_descendants} \
+                         descendant commits"
+                    )?;
+                }
```

Comment:
I'm not sure this "skipped rebasing" message is very useful but I don't feel strongly at all. To me, the "rebased N descendant commits" messages are useful mostly so you can notice if there are many more rebased commits than you expected, which may indicate that you passed the wrong commit as argument, for example. In this case, it seems like the "Changes to push to origin:" stuff is much more likely to help you spot an error.

So you have a weak vote for removing it from me, but feel free to keep it.

Actually, do you think we can skip calling `sign_commits_before_push()` completely when `--dry-run` is passed? Maybe it then becomes simple enough that it can be shared with the `jj sign` command @pylbrecht is working on in #4747.

Reply:
Yeah, I was initially thinking we might want to check the total number of descendants rebased as part of the dry run, but I agree that it's probably not the most useful and just the changes to push (or the number of commits to sign is enough).